tests/unit_tests/converse/test_converse_adapter.py

```diff
@@ -0,0 +1,232 @@
+from aidial_adapter_bedrock.dial_api.request import ModelParameters
+from aidial_adapter_bedrock.llm.converse.adapter import ConverseAdapter
+from aidial_adapter_bedrock.llm.converse.types import (
+    ConverseMessage,
+    ConverseParams,
+    ConverseRole,
+    ConverseTextPart,
+    ConverseToolResultPart,
+    ConverseToolUseConfig,
+    ConverseToolUsePart,
+    InferenceConfig,
+)
+from aidial_adapter_bedrock.llm.errors import ValidationError
+from aidial_adapter_bedrock.llm.tools.tools_config import ToolsConfig
+from aidial_adapter_bedrock.utils.list_projection import ListProjection
+
+
+async def _input_tokenizer_factory(_deployment, _params):
+    async def _test_tokenizer(_messages) -> int:
+        return 100
+
+    return _test_tokenizer
+
+
+@dataclass
+class TestCase:
+    __test__ = False
+    name: str
+    messages: List[Message]
+    params: ModelParameters
+    expected_output: ConverseParams | None = None
+    expected_error: type[Exception] | None = None
+
+
+default_inference_config = InferenceConfig(stopSequences=[])
+TEST_CASES = [
+    TestCase(
+        name="plain_message",
+        messages=[Message(role=Role.USER, content="Hello, world!")],
+        params=ModelParameters(tool_config=None),
+        expected_output=ConverseParams(
+            inferenceConfig=default_inference_config,
+            messages=ListProjection(
+                list=[
+                    (
+                        ConverseMessage(
+                            role=ConverseRole.USER,
+                            content=[ConverseTextPart(text="Hello, world!")],
+                        ),
+                        {0},
+                    )
+                ]
+            ),
+        ),
+    ),
+    TestCase(
+        name="system_message",
+        messages=[
+            Message(role=Role.SYSTEM, content="You are a helpful assistant."),
+            Message(role=Role.USER, content="Hello!"),
+        ],
+        params=ModelParameters(tool_config=None),
+        expected_output=ConverseParams(
+            inferenceConfig=default_inference_config,
+            system=[ConverseTextPart(text="You are a helpful assistant.")],
+            messages=ListProjection(
+                list=[
+                    (
+                        ConverseMessage(
+                            role=ConverseRole.USER,
+                            content=[ConverseTextPart(text="Hello!")],
+                        ),
+                        {1},
+                    )
+                ]
+            ),
+        ),
+    ),
+    TestCase(
+        name="system_message_after_user",
+        messages=[
+            Message(role=Role.SYSTEM, content="You are a helpful assistant."),
+            Message(role=Role.USER, content="Hello!"),
+            Message(role=Role.SYSTEM, content="You are a helpful assistant."),
+        ],
+        params=ModelParameters(tool_config=None),
+        expected_error=ValidationError,
```

Comment:
Check for a particular error message?

Reply:
Done, wrapped in 
```python
@dataclass
class ExpectedException:
    type: type[Exception]
    message: str
``` similar to `test_chat_completion` test